build/PHPStan/Build/MemoizationPropertyRule.php

```diff
@@ -0,0 +1,182 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Build;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\AssignOp\Coalesce;
+use PhpParser\Node\Expr\BinaryOp\Identical;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\StaticPropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\Stmt\If_;
+use PhpParser\Node\Stmt\Return_;
+use PhpParser\Node\VarLikeIdentifier;
+use PHPStan\Analyser\Scope;
+use PHPStan\File\FileHelper;
+use PHPStan\Node\InClassMethodNode;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use function count;
+use function dirname;
+use function is_string;
+use function sprintf;
+use function str_starts_with;
+use function strcasecmp;
+
+/**
+ * @implements Rule<InClassMethodNode>
+ */
+final class MemoizationPropertyRule implements Rule
+{
+
+	public function __construct(private FileHelper $fileHelper, private bool $skipTests = true)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return InClassMethodNode::class;
```

Comment:
I think the rule should apply to `If_`, not to a method body.

Reply:
Ok, it has been reflected in the rule.